src/linear_algebra/basic.lean

```diff
@@ -1029,6 +1029,20 @@ begin
   rw [range_comp, submodule.map_neg, submodule.map_id],
 end
 
+/-- A linear map version of `add_monoid_hom.eq_locus` -/
+def eq_locus (f g : M →ₛₗ[τ₁₂] M₂) : submodule R M :=
+{ carrier := {x | f x = g x},
+  smul_mem' := λ r x (hx : _ = _), show _ = _,
+    by simpa only [linear_map.map_smulₛₗ] using congr_arg ((•) (τ₁₂ r)) hx,
+  .. f.to_add_monoid_hom.eq_mlocus g.to_add_monoid_hom }
```

Comment:
I was expecting a few lemmas along the lines of `eq_locus f f = \top`, apparently they don't exist for `add_monoid_hom`s either so I'm also OK with adding those in for monoid homs, ring homs and linear maps at the same time.

Reply:
Sure, I added those three lemmas.